app/scripts/controllers/HomeCtrl.js

```diff
@@ -16,14 +22,26 @@
         };
         
         
+        // invoked by click on Create new room button
         this.setCurrentRoom = function(room) {
             this.currentRoom = room;
             this.messages = Message.getByRoomId(this.currentRoom.$id);
         };
+        
+        
+        // invoked by click on 'send' button 
+        this.sendMessage = function () {
+            this.newMessage.roomId = this.currentRoom.$id;

```

Comment:
Using `this.newMessage` was rather confusing to me. I had to ask where it came from and why we needed to access it using `this`.  Then I realized you had created that property in the view using `ng-model`, and also that you are implying here that when you call `Message.send(this.newMessage)` that `this.newMessage` will include `this.newMessage.content`, which is not at all obvious to any future developer--it feels like magic. I would prefer to see everything set explicitly, something like:

```
this.sendMessage = function () {
    var message = {
        roomId: this.currentRoom.$id,
        userName: this.currentUser,
        sentAt: firebase.database.ServerValue.TIMESTAMP,
        content: this.newMessage.content
    };

    Message.send(message);
    this.newMessage.content = '';
};
```